compiler/rustc_ty_utils/src/instance.rs

```diff
@@ -97,7 +97,11 @@ fn resolve_associated_item<'tcx>(
 ) -> Result<Option<Instance<'tcx>>, ErrorGuaranteed> {
     debug!(?trait_item_id, ?param_env, ?trait_id, ?rcvr_substs, "resolve_associated_item");
 
-    let trait_ref = ty::TraitRef::from_method(tcx, trait_id, rcvr_substs);
+    let trait_ref = ty::TraitRef::from_method(
+        tcx,
+        trait_id,
+        tcx.normalize_erasing_regions(param_env, rcvr_substs),
```

Comment:
I think I would prefer if we find what parent function isn't normalizing the rcvr_substs and normalize there instead

Reply:
I've fixed that.